state_machine/state_engine.py

```
@@ -121,45 +122,50 @@ def set_phase(self, phase: MissionPhase) -> Dict[str, Any]:
+        try:
             try:
-                MissionPhaseValidator.validate_transition(
-                    self.current_phase.value, validated_phase.value
-                )
+                # Validate phase string and type
+                if isinstance(phase, MissionPhase):
+                    target_phase_str = phase.value
+                elif isinstance(phase, str):
+                    target_phase_str = phase
+                else:
+                     raise ValidationError(f"Invalid phase type: {type(phase)}")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)